src/main/java/de/triology/cas/services/CesServicesManagerStageProductive.java

```diff
@@ -89,21 +81,28 @@ private void synchronizeServicesWithRegistry() {
      */
     private void registerChangeListener() {
         logger.debug("entered registerChangeListener");
-        try {
-            registry.addDoguChangeListener(()-> {
-                logger.debug("registered change in /dogu");
-                synchronizeServicesWithRegistry();
-            });
-        } catch (RegistryException ex) {
-            logger.error("failed to synchronizeServicesWithRegistry service", ex);
-        }
+        registry.addDoguChangeListener(()-> {
+            logger.debug("registered change in /dogu");
+            synchronizeServicesWithRegistry();
+        });
     }
 
     /**
      * Creates and registers a new service for an given name
      */
-    private void addNewService(String name) {
-        addNewService(name, "https://" + fqdn + "(:443)?/" + name + "(/.*)?");
+    void addNewService(String name) {
+        String serviceId = "https://" + fqdn + "(:443)?/" + name + "(/.*)?";
+        if (registry instanceof RegistryEtcd){
```

Comment:
Isn't getCasLogoutUri() a method all Registries should implement? 
This would eliminate this check and the casting two lines below.